### PR TITLE
Feature profile picture - Implemented profile picture system

### PR DESCRIPTION
**Pull request by AlouchLaBouche**
It is now possible to add a profile picture when creating an account 🚀 !
### This PR includes: 
Additions to `AccountDetails.kt`, to be able to handle images. The notable ones are:
 - `ImageRepository` that is now passed as paramter to AccountDetails. 
 - A `photo picker` that enables the user to fetch images from his gallery, see the [documentation](https://developer.android.com/training/data-storage/shared/photopicker) to learn more.  
 - And a `ProfilePictureWithRemoveIcon` composable for the preview of the image.
 
 I've also refactored the user repository as it had a missing method in the interface. 

A modification to the `UserProfile.kt` file to crop the profile picture correctly.

### CODE REVIEW

**Comment by AlouchLaBouche** [comment]
Sadly this PR doesn't include any new tests as it is very diffcult to mock the `photo picker` functionality provided by `JetPackCompose`. Since the whole composable is out of the scope of our project, I have no idea how to mock a user choosing a profile picture from his gallery 😥.

**Comment by oskar-codes** [comment]
ESPECE DE FILS DE PUTE EKLE EST EN DRAFT COMMENT TU VEUX QU'ON LA REVIEW T'ES CON A CHIER VA TE FAIRE FOUTRE PTN ESPECE DE FRANCO-RUSSE DE MERDE DE MONGOL RETOURNE DANS TON PAYS TON LINE CONTRIBUTION PUE LA MERDE TU CONTRIBUES AUTANT QUE MON CUL SUR TON CLAVIER MEME MA GRAND-MERE AVEC ALZHEIMER, DEMENTIA, PARKINSONS SCOLIOSE ET QUI EST MORTE CONTRIBUE PLUS

File: app/src/androidTest/java/com/android/unio/ui/authentication/AccountDetailsTest.kt [hunk 1]
```diff
@@ -122,6 +130,22 @@ class AccountDetailsTest {
     composeTestRule.onNodeWithTag("AccountDetailsInterestChip: 1").assertExists()
   }
 
+  @Test
+  fun testAddingSocialsCorrectlyModifiesTheFlowRow() {
```
Comment (Zafouche) [inline app/src/androidTest/java/com/android/unio/ui/authentication/AccountDetailsTest.kt:134]: Good test

File: app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt [hunk 2]
```diff
@@ -54,11 +67,28 @@ import com.google.firebase.Firebase
 import com.google.firebase.auth.auth
 import kotlinx.coroutines.flow.MutableStateFlow
 
+//
+//
+// @Preview
```
Comment (Zafouche) [inline app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt]: Don't know if a preview should be pushed, even if it is commented. Perhaps you can just store it locally on another file you don't push.

File: app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt [hunk 3]
```diff
@@ -75,7 +105,20 @@ fun AccountDetails(
   val interests by interestsFlow.collectAsState()
   val socials by userSocialsFlow.collectAsState()
 
+  val profilePictureUri = remember { mutableStateOf<Uri>(Uri.EMPTY) }
+
+  val pickMedia =
+      rememberLauncherForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+        if (uri != null) {
+          Log.d("PhotoPicker", "Selected URI: $uri")
```
Comment (Zafouche) [inline app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt:97]: Don't forget to remove debug logs!

File: app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt [hunk 4]
```diff
@@ -75,7 +105,20 @@ fun AccountDetails(
...
+          profilePictureUri.value = uri
+        } else {
+          Log.d("PhotoPicker", "No media selected")
```
Comment (Zafouche) [inline app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt:100]: This one as well

File: app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt [hunk 5]
```diff
@@ -84,6 +127,33 @@ fun AccountDetails(
     navigationAction.navigateTo(Screen.WELCOME)
     return
   }
+  val userId = Firebase.auth.currentUser?.uid
+
+  val createUser = { uri: String ->
```
Comment (Zafouche) [inline app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt:116]: Well defined callback

**Review by Zafouche — APPROVED** [review]
Отличная работа Алексей! Не забудьте оптимизировать импорт и после этого запустить «ktfmtFormat».

Really nothing to say, except that the commented preview shouldn't be pushed and that you forgot to remove a couple of debug Logs! Thank you for your dedication, брат.

File: app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt (hunk 3 above)
Comment (AlouchLaBouche) [reply app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt:97]: Good spot, I've forgot about those, thanks партнер 💪. I'll remove them in a giffy.

**Pull request merged**